src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs

```diff
@@ -753,6 +751,50 @@ private string ConfigureDatabase(InitializeViewRequestParams viewParams, Databas
                             }
                         }
 
+                        if (!viewParams.IsNewObject && database.Filegroups != null)
+                        {
+                            HashSet<string> fileGroupToRemove = new HashSet<string>(prototype.Filegroups.Select(file => file.Name));
```

Comment:
Since we aren't calling `Contains` on this Set until after we've trimmed its elements, could you start with this HashSet being empty and then add items to it as necessary rather than removing them? That would reduce our overall memory usage with this HashSet. For example, if you only ended up needing to track one FileGroup here to set `Removed = true` on, but the HashSet started out containing a hundred others that end up getting removed from the HashSet, then that was memory usage you didn't need.

Reply:
Achieved it without using any extra set, and modified the Files save logic too. Thanks